input_rpc.go

```diff
@@ -24,22 +25,40 @@ const (
 // The handlers maintain full compatibility with existing RPC interface
 // while providing significant latency improvements for input events.
 
+// Ultra-fast input RPC structures for zero-allocation parsing
+// Bypasses float64 conversion by using typed JSON unmarshaling
+
+// InputRPCRequest represents a specialized JSON-RPC request for input methods
+// This eliminates the map[string]interface{} overhead and float64 conversions
+type InputRPCRequest struct {
+	JSONRPC string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	ID      any    `json:"id,omitempty"`
+	// Union of all possible input parameters - only relevant fields are populated
+	// Fields ordered for optimal 32-bit alignment: slice pointers, 2-byte fields, 1-byte fields
+	Params struct {
```

Comment:
This means each input RPC message is carrying around a lot of extra bytes (for the unused parts of the struct), can't we use a custom serializer/deserializer for these that only transfers the actual bytes?

Reply:
That would be really cool, indeed... but we will probably lose on the maintainability front. I was thinking about that actually, and I think some work in that direction (having RPC use a custom, less-overhead serializer is under way by @ym if I recall correctly? or at least it's planned - I think this should align with that, eventually)

Reply:
Yes, we have a channel for frequently transmitted messages using custom encoding. You can find it here: 
https://github.com/jetkvm/kvm/commit/bcc307b147be563391cd88eee70d7daf5a67a679.

It was merged into the dev branch last week.

Reply:
File removed